src/widgets/arc/lv_arc.c

```diff
@@ -655,6 +655,24 @@ static void lv_arc_event(const lv_obj_class_t * class_p, lv_event_t * e)
             return;
         }
 
+        /*Calculate the angle of the pressed point*/
+        lv_value_precise_t angle = lv_atan2(info->point->y - p.y, info->point->x - p.x);
+        angle -= arc->rotation;
+        angle -= arc->bg_angle_start;  /*Make the angle relative to the start angle*/
+
+        /* If we click near the bg_angle_start the angle will be close to 360° instead of a small angle */
+        if(angle < 0) angle += 360;
```

Comment:
```suggestion
        while(angle < 0) angle += 360;
```

I was able to break your example by adding
```c
    lv_arc_set_rotation(arc1, 355);
    lv_arc_set_rotation(arc2, 355);
```

Maybe also change it in the other place this logic is.

A "better" (but less readable) alternative would be `angle = (angle % 360 + 360) % 360;` according to ChatGPT.

Reply:
Thanks for catching that!

I've pushed a new commit to ensure the angle remains between 0 and 360 in several instances. After re-testing, it seems to have resolved the issue when adding large positive or negative angles.

There may be other places where this could occur, such as in lv_arc_set_start_angle. However, since I'm still relatively new to LVGL, I'm not entirely sure how it internally handles angles outside the 0-360 range.